### PR TITLE
React Native upgrade 0.73.4

### PR DESCRIPTION
**Pull request by chennara**

### CODE REVIEW

File: android/app/src/main/java/com/reactnative_uitpas/MainActivity.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package com.publiq.uitpas
+
+import com.facebook.react.ReactActivity
+import com.facebook.react.ReactActivityDelegate
+import com.facebook.react.defaults.DefaultNewArchitectureEntryPoint.fabricEnabled
+import com.facebook.react.defaults.DefaultReactActivityDelegate
+
+class MainActivity : ReactActivity() {
```
Comment (samvanhoeyicapps) [inline android/app/src/main/java/com/reactnative_uitpas/MainActivity.kt:8]: where is the onCreate with the splashScreen

File: .nvmrc [hunk 2]
```diff
@@ -1 +1 @@
-16.13.1
\ No newline at end of file
+18
```
Comment (samvanhoeyicapps) [inline .nvmrc:1]: does react native not work with 20?

File: android/app/src/main/java/com/reactnative_uitpas/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    override fun getMainComponentName(): String = "reactnative_uitpas"
+
+    /**
+     * Returns the instance of the [ReactActivityDelegate]. We use [DefaultReactActivityDelegate]
+     * which allows you to enable New Architecture with a single boolean flags [fabricEnabled]
+     */
+    override fun createReactActivityDelegate(): ReactActivityDelegate =
+        DefaultReactActivityDelegate(this, mainComponentName, fabricEnabled)
+
+    override fun onCreate(savedInstanceState: Bundle?) {
```
Comment (samvanhoeyicapps) [inline android/app/src/main/java/com/reactnative_uitpas/MainActivity.kt:25]: maybe also look at this thread? The saved instance might be the one that causes the crash rate https://github.com/software-mansion/react-native-screens/issues/17#issuecomment-989623502

File: android/build.gradle [hunk 4]
```diff
@@ -1,61 +1,39 @@
...
     ext {
-        buildToolsVersion = "31.0.0"
-        minSdkVersion = 21
-        compileSdkVersion = 31
-        targetSdkVersion = 33
-        kotlin_version = '1.6.10'
+        buildToolsVersion = "34.0.0"
+        minSdkVersion = 26
+        compileSdkVersion = 34
+        targetSdkVersion = 34
```
Comment (samvanhoeyicapps) [inline android/build.gradle:8]: that's a nice bump 👍

File: babel.config.js [hunk 5]
```diff
@@ -1,8 +1,4 @@
 module.exports = {
-  plugins: [
-    ['react-native-reanimated/plugin', { // reanimated must be the last package
-      globals: ['__scanCodes'],
```
Comment (samvanhoeyicapps) [inline babel.config.js:4]: no need for the globals anymore?

File: src/_components/navigation/NavigationBar.tsx [hunk 6]
```diff
@@ -15,10 +15,22 @@ const NavigationBar: FC = () => {
...
+        <Rect fill={theme.palette.neutral['100']} height="100%" mask="url(#cutout-border)" width="100%" x={0} y={0} />
+      )}
+      {/* Only on Android, this will represent the border of the navigationbar */}
+      <Rect
+        fill={theme.palette.neutral['0']}
+        height="100%"
+        mask="url(#cutout)"
+        width="100%"
+        x={0}
+        y={Platform.OS === 'android' ? 1 : 0}
```
Comment (samvanhoeyicapps) [inline src/_components/navigation/NavigationBar.tsx:31]: did you change the line length?

File: src/_context/AuthenticationContext.tsx [hunk 7]
```diff
@@ -86,7 +86,7 @@ const AuthenticationProvider: FC<PropsWithChildren> = ({ children }) => {
   }, [client, setIsAuthenticated, logout, setAccessToken, setUser, setIsInitialized]);
 
   useAppState({
-    onForeground: async () => {
+    onForeground: () => {
```
Comment (samvanhoeyicapps) [inline src/_context/AuthenticationContext.tsx:89]: 👍

File: package.json [hunk 8]
```diff
@@ -34,62 +34,69 @@
...
-    "lottie-ios": "3.2.3",
-    "lottie-react-native": "5.1.3",
-    "react": "18.1.0",
+    "lottie-ios": "3.4.4",
+    "lottie-react-native": "5.1.6",
+    "react": "18.2.0",
     "react-hook-form": "^7.48.2",
     "react-i18next": "^12.0.0",
-    "react-native": "0.70.10",
+    "react-native": "0.73.4",
```
Comment (samvanhoeyicapps) [inline package.json:42]: 73.5 was just released maybe already add that

File: package.json
Comment (samvanhoeyicapps) [inline package.json:1]: does yarn ios still work with configuration Debug?

Comment (samvanhoeyicapps) [reply package.json:1]: I think it should be mode

**Review by samvanhoeyicapps — CHANGES_REQUESTED** [review]

File: babel.config.js (hunk 5 above)
Comment (chennara) [reply babel.config.js:4]: This was needed for the `vision-camera-code-scanner` setup, which we don't use anymore.

**Review by samvanhoeyicapps — APPROVED** [review]

**Pull request merged**